File: Week03/LeetCode-2032.cpp
class Solution {
public:
    void InsersectSet(vector<int> &nums1, vector<int> &nums2, vector<int> &result){
        int i=0, j=0;
        while (i<nums1.size() and j<nums2.size()){
            if (nums1[i] == nums2[j]){
                result.push_back(nums1[i]);
                i++;
                j++;
                                
            } else if (nums1[i] < nums2[j]) i++;
            else j++;
        } // end-while
    } // end-InsersectSet

    vector<int> twoOutOfThree(vector<int>& nums1, vector<int>& nums2, vector<int>& nums3) {
        // Sort all arrays
        std::sort(nums1.begin(), nums1.end());
        std::sort(nums2.begin(), nums2.end());
        std::sort(nums3.begin(), nums3.end());
        
        vector<int> result;
        InsersectSet(nums1, nums2, result);
        InsersectSet(nums1, nums3, result);
        InsersectSet(nums2, nums3, result);
        
        if (result.size() <= 1) return result;
        
        // Eliminate the duplicates from the result
        std::sort(result.begin(), result.end());
        auto iter = std::unique(result.begin(), result.end());
        result.resize(iter-result.begin());
        
        return result;                
    } // end-twoOutOfThree
};

File: Week03/LeetCode-2032.py
class Solution:
    def InsersectSet(self, nums1, nums2, result):
        i, j = 0, 0
        while i<len(nums1) and j<len(nums2):
            if (nums1[i] == nums2[j]):
                result.append(nums1[i])
                i += 1
                j += 1
            elif nums1[i] < nums2[j]: i+= 1
            else: j += 1

    def twoOutOfThree(self, nums1: List[int], nums2: List[int], nums3: List[int]) -> List[int]:
        # Sort the arrays
        nums1.sort()
        nums2.sort()
        nums3.sort()
        
        result = []
        self.InsersectSet(nums1, nums2, result)
        self.InsersectSet(nums1, nums3, result)
        self.InsersectSet(nums2, nums3, result)
                        
        if len(result)<=1: return result
        
        # Eliminate the duplicates from the result
        result.sort()
        k = 0
        for i in range(1, len(result)):
            if result[i] != result[k]:
                k += 1
                result[k] = result[i]
        return result[:k+1]